File: src/test/java/io/polygloat/service/DbPopulatorTest.java
package io.polygloat.service;

import io.polygloat.configuration.polygloat.PolygloatProperties;
import io.polygloat.development.DbPopulatorReal;
import io.polygloat.exceptions.NotFoundException;
import io.polygloat.model.ApiKey;
import io.polygloat.model.UserAccount;
import io.polygloat.repository.RepositoryRepository;
import io.polygloat.repository.UserAccountRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.testng.AbstractTransactionalTestNGSpringContextTests;
import org.springframework.transaction.PlatformTransactionManager;
import org.springframework.transaction.annotation.Transactional;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import javax.persistence.EntityManager;
import java.util.Optional;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
public class DbPopulatorTest extends AbstractTransactionalTestNGSpringContextTests {

    @Autowired
    DbPopulatorReal populator;

    @Autowired
    UserAccountRepository userAccountRepository;

    @Autowired
    RepositoryRepository repositoryRepository;

    @Autowired
    PlatformTransactionManager transactionManager;

    @Autowired
    ApiKeyService apiKeyService;

    @Autowired
    EntityManager entityManager;

    @Autowired
    PolygloatProperties polygloatProperties;


    private UserAccount userAccount;

    @BeforeMethod
    public void setup() {
        populator.autoPopulate();
        userAccount = userAccountRepository.findByUsername(polygloatProperties.getAuthentication().getInitialUsername()).orElseThrow(NotFoundException::new);
    }

    @Test
    @Transactional
    void createsUser() {
        assertThat(userAccount.getName()).isEqualTo(polygloatProperties.getAuthentication().getInitialUsername());
    }

    @Test
    @Transactional
    void createsRepository() {
        entityManager.refresh(userAccount);
        assertThat(userAccount.getCreatedRepositories().size()).isPositive();
    }

    @Test
    @Transactional
    void createsApiKey() {
        Optional<ApiKey> key = apiKeyService.getAllByUser(userAccount).stream().findFirst();
        assertThat(key).isPresent();
        ApiKey apiKey = key.get();
        assertThat(apiKey.getKey()).isEqualTo("this_is_dummy_api_key");
    }
}

File: backend/app/src/test/kotlin/io/tolgee/service/DbPopulatorTest.kt
package io.tolgee.service

import io.tolgee.configuration.tolgee.TolgeeProperties
import io.tolgee.development.DbPopulatorReal
import io.tolgee.exceptions.NotFoundException
import io.tolgee.model.UserAccount
import io.tolgee.repository.UserAccountRepository
import io.tolgee.service.security.ApiKeyService
import io.tolgee.testing.AbstractTransactionalTest
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.transaction.annotation.Transactional

@SpringBootTest
class DbPopulatorTest : AbstractTransactionalTest() {
  @Autowired
  lateinit var populator: DbPopulatorReal

  @Autowired
  lateinit var userAccountRepository: UserAccountRepository

  @Autowired
  lateinit var apiKeyService: ApiKeyService

  @Autowired
  lateinit var tolgeeProperties: TolgeeProperties

  lateinit var userAccount: UserAccount

  @BeforeEach
  fun setup() {
    populator.autoPopulate()
    userAccount =
      userAccountRepository
        .findByUsername(tolgeeProperties.authentication.initialUsername)
        .orElseThrow { NotFoundException() }
  }

  @Test
  @Transactional
  fun createsUser() {
    Assertions.assertThat(userAccount.name).isEqualTo(tolgeeProperties.authentication.initialUsername)
  }

  @Test
  @Transactional
  fun createsApiKey() {
    val key = apiKeyService.getAllByUser(userAccount.id).stream().findFirst()
    Assertions.assertThat(key).isPresent
    Assertions.assertThat(key.get().key).isEqualTo(null)
  }
}
